infrastructure/tooling/src/importer/importer.js

```diff
@@ -1,49 +1,149 @@
+};
+
+const makeTableOperations = async ({operations, config, monitor, credentials, db}) => {
+  const tableNames = [];
   for (let path of glob.sync('services/*/azure.yml', {cwd: REPO_ROOT})) {
     const azureYml = await readRepoYAML(path);
     for (let t of azureYml.tables || []) {
-      if (ALLOWED_TABLES.includes(t)) {
-        tables.push(t);
+      tableNames.push(t);
+    }
+  }
+
+  // TODO: apply ALLOWED_TABLES
+
+  // support truncating each table exactly once
+  const truncated = {};
+  const truncateTable = tableName => {
+    if (!truncated[tableName]) {
+      truncated[tableName] = (async () => {
+        const pgTable = postgresTableName(tableName);
+        await db._withClient('admin', async client => {
+          await client.query(`truncate ${pgTable}`);
+        });
+      })();
+    }
+    return truncated[tableName];
+  };
+
+  for (let tableName of tableNames) {
+    if (LARGE_TABLES.includes(tableName)) {
+      for (let range of TASKID_RANGES) {
+        operations.add(new TableOperation({tableName, range, db, credentials, truncateTable}));
       }
+    } else {
+      operations.add(new TableOperation({tableName, db, credentials, truncateTable}));
     }
   }
+};
 
-  async function importTable(tableName, tableParameters = {}, rowsProcessed = 0, utils) {
-    for await (let result of buffer(
-      readAzureTableInChunks({
-        azureCreds: credentials.azure,
-        tableName,
-        utils,
-        tableParams: tableParameters,
-        rowsProcessed,
-      }),
-      5,
-    )) {
-      const { entities, count } = result;
+class TableOperation extends Operation {
+  constructor({tableName, range, db, credentials, truncateTable}) {
+    super({title: `${tableName} table`});
+    this.tableName = tableName;
+    this.db = db;
+    this.credentials = credentials;
+    this.truncateTable = truncateTable;
+
+    if (range) {
+      const [from, to] = range;
+      if (from && to) {
+        this.filter = `PartitionKey ${Table.Operators.GreaterThanOrEqual} ${Table.Operators.string(from)} ${Table.Operators.And} PartitionKey ${Table.Operators.LessThan} ${Table.Operators.string(to)}`;
+        this.title = `${this.title} ${from} ≤ taskId < ${to}`;
+      } else if (from && !to) {
+        this.filter = `PartitionKey ${Table.Operators.GreaterThanOrEqual} ${Table.Operators.string(from)}`;
+        this.title = `${this.title} ${from} ≤ taskId`;
+      } else if (!from && to) {
+        this.filter = `PartitionKey ${Table.Operators.LessThan} ${Table.Operators.string(to)}`;
+        this.title = `${this.title} taskId < ${to}`;
+      } else {
+        throw new Error('weird range');
+      }
+    } else {
+      this.filter = undefined;
+    }
+  }
 
-      await writeToPostgres(tableName, entities, db, ALLOWED_TABLES);
+  async run() {
+    await this.truncateTable(this.tableName);
+    this.rowsProcessed(0);
 
-      rowsProcessed = count;
+    const BATCH_SIZE = 1000;
+    const BUFFER_SIZE = 2000;
```

Comment:
I think the buffer size was set to 5. Looking at the latest CPU usage from the db, we were hitting 40-60%. Do we really want to up that number to 2000?

Reply:
That's now a number of rows, not a number of pages, so it's actually a reduction in buffer size from 5 pages = 5000 rows.  The buffer never seems to fill (that's the buffer % in the output), so I don't think the size is particularly important.